src/main/java/scala/compat/java8/runtime/LambdaDeserializer.scala

```diff
@@ -0,0 +1,112 @@
+   *    not stored in `SerializedLambda`, so we can't reconstitute them.
+   *  - No additional bridge methods are passed to `altMetafactory`. Again, these are not stored.
+   *
+   * Note: The Java compiler
+   *
+   * @param lookup      The factory for method handles. Must have access to the implementation method, the
+   *                    functional interface class, and `java.io.Serializable` or `scala.Serializable` as
+   *                    required.
+   * @param serialized  The lambda to deserialize. Note that this is typically created by the `readResolve`
+   *                    member of the anonymous class created by `LambdaMetaFactory`.
+   * @return            An instance of the functional interface
+   */
+  def deserializeLambda(lookup: MethodHandles.Lookup, serialized: SerializedLambda): AnyRef = {
+    def slashDot(name: String) = name.replaceAll("/", ".")
+    val loader = lookup.lookupClass().getClassLoader
+    val implClass = loader.loadClass(slashDot(serialized.getImplClass))
+
+    def makeCallSite: CallSite = {
+      import serialized._
+      def parseDescriptor(s: String) =
+        MethodType.fromMethodDescriptorString(s, loader)
+
+      val funcInterfacesSignature = parseDescriptor(getFunctionalInterfaceMethodSignature)
+      val methodType: MethodType = funcInterfacesSignature
+      val instantiated = parseDescriptor(getInstantiatedMethodType)
+      val implMethodSig = parseDescriptor(getImplMethodSignature)
+
+      val from = implMethodSig.parameterCount() - funcInterfacesSignature.parameterCount()
+      val to = implMethodSig.parameterCount()
+      val functionalInterfaceClass = loader.loadClass(slashDot(getFunctionalInterfaceClass))
+      var invokedType: MethodType =
+        implMethodSig.dropParameterTypes(from, to)
+          .changeReturnType(functionalInterfaceClass)
+
+      val implMethod: MethodHandle = try {
+        getImplMethodKind match {
+          case MethodHandleInfo.REF_invokeStatic =>
+            lookup.findStatic(implClass, getImplMethodName, implMethodSig)
+          case MethodHandleInfo.REF_invokeVirtual =>
+            invokedType = invokedType.insertParameterTypes(0, implClass)
+            lookup.findVirtual(implClass, getImplMethodName, implMethodSig)
+          case MethodHandleInfo.REF_invokeSpecial =>
+            invokedType = invokedType.insertParameterTypes(0, implClass)
+            lookup.findSpecial(implClass, getImplMethodName, implMethodSig, implClass)
+        }
+      } catch {
+        case e: ReflectiveOperationException =>
+          throw new IllegalArgumentException("Illegal lambda deserialization", e)
+      }
+      val FLAG_SERIALIZABLE = 1
+      val FLAG_MARKERS = 2
+      val flags: Int = FLAG_SERIALIZABLE | FLAG_MARKERS
+      val markerInterface: Class[_] = if (functionalInterfaceClass.getName.startsWith("scala.Function"))
+        loader.loadClass("scala.Serializable")
+      else
+        loader.loadClass("java.io.Serializable")
+
```

Comment:
So `FLAG_SERIALIZABLE` seems to already [add the `java.io.Serializable` interface](https://docs.oracle.com/javase/8/docs/api/java/lang/invoke/LambdaMetafactory.html#altMetafactory-java.lang.invoke.MethodHandles.Lookup-java.lang.String-java.lang.invoke.MethodType-java.lang.Object...-).

We should check what happens if we also add `scala.Serializable` as a marker interface - do we get both? Is that OK?